### PR TITLE
Add plateau learning rate schedule, add trigger for loss and score

### PR DESCRIPTION
**Pull request by SeaOfOcean**
## What changes were proposed in this pull request?

Add plateau learning rate schedule, add trigger for loss and score
Add two float variable loss and score to driverState and optimState
If there are multiple validation method, only the first score will be recorded.

## How was this patch tested?

unit test

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/Trigger.scala [hunk 1]
```diff
@@ -99,5 +99,29 @@ object Trigger {
...
+   * A trigger that triggers an action when training loss less than "min" loss
+   * @param min min loss
+   */
+  def minLoss(min: Float): Trigger = {
+    new Trigger() {
+      override def apply(state: Table): Boolean = {
+        state[Float]("Loss") < min
+      }
+    }
+  }
```
Comment (shane-huang) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/Trigger.scala]: May not be addressed in this pr, but we could give it a thought of below cases: 
- we are at some point, where the loss is below "min" and will actually continue to drop if we train longer
- we are at some point, where the loss is just a little bit above the "min" but never goes below "min". It trains forever.

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/DistriOptimizer.scala [hunk 2]
```diff
@@ -262,6 +264,10 @@ object DistriOptimizer {
           parameters.gradientPartition.div(ev.fromType(finishedModelNum))
           modelCache.optimMethod.state.update("epoch", driverState[Int]("epoch"))
           modelCache.optimMethod.state.update("neval", driverState[Int]("neval"))
+          modelCache.optimMethod.state.update("Loss", driverState[Float]("Loss"))
```
Comment (zhangxiaoli73) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/DistriOptimizer.scala]: If you don't use Loss and score in your learningRate schedule, I think you don't have to pass then to modelCache.optimMethod

Comment (SeaOfOcean) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/DistriOptimizer.scala]: To support Loss trigger and score Trigger, we need record them

**Comment by SeaOfOcean** [comment]
pr validate pass

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/DistriOptimizer.scala (hunk 2 above)
Comment (zhangxiaoli73) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/DistriOptimizer.scala]: ok

**Comment by zhichao-li** [comment]
LGTM
